### PR TITLE
test: surface compute sanitizer logs

### LINKED ISSUE #1337: Make the error log more friendly when Compute Sanitizer fails the build.
This is a follow-up for PR https://github.com/NVIDIA/spark-rapids-jni/pull/1321.

The below is a typical error log when all unit tests pass but the Compute Sanitizer catches some errors and fails the build process.
It is not easy to locate the failing tests according to the log. We may investigate it to see if it could be improved.
```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3:42.375s
[INFO] Finished at: Thu Aug 10 09:25:37 UTC 2023
[INFO] Final Memory: 62M/1290M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project spark-rapids-jni: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/liangcail/work/projects/on_github/spark-rapids-jni/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] Error occurred in starting fork, check output in log
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: Error occurred in starting fork, check output in log
[ERROR] at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:624)
[ERROR] at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
[ERROR] at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
[ERROR] at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
[ERROR] at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
[ERROR] at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
[ERROR] at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
[ERROR] at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
[ERROR] at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
[ERROR] at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
[ERROR] at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
[ERROR] at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
[ERROR] at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
[ERROR] at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
[ERROR] at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
```

### PR DESCRIPTION
**Pull request by WilliamK112**
## Summary

- isolate each Compute Sanitizer wrapper invocation with a unique log basename so parallel Surefire forks do not consume sibling logs
- write sanitizer reports into `target/surefire-reports` when Maven runs the wrapper
- let Maven print the first 80 lines of every sanitizer report to the normal build output after the tests, instead of writing excerpts to the forked JVM's stderr/dumpstream
- fail the Maven build when a report does not contain `ERROR SUMMARY: 0 errors`
- remove stale sanitizer reports before each test run
- forward TERM/INT/HUP from the wrapper to the running Compute Sanitizer process while preserving its exit behavior
- document the report location and Maven output behavior

Closes #1337.

## Validation

Revalidated on 2026-08-09 after rebasing onto current `main`:

- `bash -n build/sanitizer-java/bin/java`
- `xmllint --noout pom.xml`
- `mvn -DskipTests -Dsubmodule.check.skip=true -DUSE_SANITIZER=ON help:effective-pom`
  - the active effective build contains one `clean-sanitizer-logs` execution and one `print-sanitizer-logs` execution
- direct `antrun:run@print-sanitizer-logs` integration checks
  - a report with `ERROR SUMMARY: 1 error` was printed and made Maven exit nonzero
  - a report with `ERROR SUMMARY: 0 errors` was printed and Maven exited successfully
- fake Compute Sanitizer wrapper checks
  - success remained quiet, exited 0, and wrote one uniquely named report to `SANITIZER_LOG_DIR`
  - failure remained quiet for Surefire capture, preserved exit 254, and wrote its report to `SANITIZER_LOG_DIR`
  - report aggregation printed the beginning rather than the end of a report and kept output bounded to 80 lines
- `git diff --check`

The previous head also passed the repository license, ShellCheck, signoff, Greptile, and pre-commit checks; those checks are rerunning for the current signed head.

### CODE REVIEW

**Comment by WilliamK112** [comment]
I reproduced the current pre-commit.ci failure locally by merging this branch onto current origin/main: the clang-format hook modified native source files introduced by the latest main after the cudf submodule sync (#4777), not this sanitizer wrapper change. I opened #4778 with only the hook-generated formatting changes, and #4778 is now passing pre-commit.ci, Greptile, and signoff. Once that lands or the same formatting is applied on main, this PR's remaining red pre-commit should be resolved by rerun/rebase.

File: build/sanitizer-java/bin/java [hunk 1]
```diff
@@ -18,8 +18,29 @@
...
+log_token="${BASHPID:-$$}_${RANDOM}_${RANDOM}"
+log_basename="sanitizer_for_pid_${log_token}"
+
+compute-sanitizer --tool memcheck \
     --launch-timeout 600 \
     --error-exitcode -2 \
-    --log-file "./sanitizer_for_pid_%p.log" \
+    --log-file "./${log_basename}_%p.log" \
     java "$@"
+status=$?
```
[the remark this answers is not in the document]
Comment (WilliamK112) [reply build/sanitizer-java/bin/java:48]: Addressed in `84895da2` by running `compute-sanitizer` as a background job, storing `sanitizer_pid`, trapping TERM/INT/HUP, forwarding the signal to that PID, and waiting for it before exiting/log-surfacing. I also added a local fake long-running `compute-sanitizer` validation: sending TERM to the wrapper makes the child receive TERM, and the wrapper exits 143 after the child is reaped. Existing failure/success fake-wrapper checks and `bash -n` still pass.

**Comment by ttnghia** [comment]
Do not dump the entire log - it may be huge. Just take the last few lines that are enough for quick inspection.

**Comment by WilliamK112** [comment]
Updated this PR for the latest review feedback:

- rebased onto current `main` after #4778 landed, so the previous unrelated clang-format pre-commit failure should be gone on rerun
- changed the sanitizer failure path to print only the last 80 lines of each matching log instead of dumping the entire file
- fixed the signal-forwarding race Greptile flagged by only applying `forwarded_status` when `wait` returned a signal-interrupted status (`>128`)

Local validation passed:
- `bash -n build/sanitizer-java/bin/java`
- fake failing/successful/signal-forwarding `compute-sanitizer` wrapper tests
- `git diff --check origin/main...HEAD`
- `uvx pre-commit run --files build/sanitizer-java/bin/java --verbose`
- `uvx pre-commit run --all-files --verbose`

**Comment by nvauto** [comment]
**NOTE: release/26.08 has been created from main. Please retarget your PR to release/26.08 if it should be included in the release.**

**Comment by WilliamK112** [comment]
Retargeted this PR to `release/26.08` after the release branch notification. I checked the branch against `upstream/release/26.08` with `git merge-tree` and revalidated the wrapper locally on 2026-07-20: syntax, diff whitespace, fake failure/success/signal-forwarding compute-sanitizer tests, shellcheck on the committed LF blob, and applicable pre-commit hooks for the changed file.

**Comment by abellina** [comment]
build

File: build/sanitizer-java/bin/java [hunk 2]
```diff
@@ -18,8 +18,54 @@
...
+trap 'forward_signal TERM 143' TERM
+trap 'forward_signal INT 130' INT
+trap 'forward_signal HUP 129' HUP
+
+compute-sanitizer --tool memcheck \
     --launch-timeout 600 \
     --error-exitcode -2 \
-    --log-file "./sanitizer_for_pid_%p.log" \
-    java "$@"
+    --log-file "./${log_basename}_%p.log" \
```
Comment (abellina) [inline build/sanitizer-java/bin/java]: I used AI to review some of this and it pointed out that the path used here may or may not be consistent with the path at `find` time later. I am not sure how often that is true, but I think it would not hurt if we made this absolute, so you'd record `WORKDIR=$PWD` then --log-file is `$WORKDIR/${log_basename}_%p.log`, and the `find` uses `$WORKDIR` as the absolute prefix. Thoughts?

**Review by abellina — COMMENTED** [review]
I have one comment and I was wondering if we can force a failure to see the output to add this to the PR. For example, find some kernel we have and then clearly write or read to memory outside of the range. compute-sanitizer should catch and report, that way we can see what it looks like.

File: build/sanitizer-java/bin/java [hunk 3]
```diff
@@ -18,8 +18,54 @@
...
+forward_signal() {
+    local signal="$1"
+    local status_after_signal="$2"
+    forwarded_status="$status_after_signal"
+    if [[ -n "$sanitizer_pid" ]]; then
+        kill "-$signal" "$sanitizer_pid" 2>/dev/null || true
+    fi
+}
+
+trap 'forward_signal TERM 143' TERM
```
Comment (abellina) [inline build/sanitizer-java/bin/java]: I'd prefer us not to hardcode the signals here, and instead use re-raising.

Lets change these traps to:
```
trap 'forward_signal TERM' TERM
trap 'forward_signal INT' INT
trap 'forward_signal HUP' HUP
```

so then `forward_signal` would be:

```
received_signal =""

# forward_signal kills the compute-sanitizer process with the signal received
forward_signal() {
    received_signal="$1"
    if [[ -n "$sanitizer_pid" ]]; then
        kill "-$received_signal" "$sanitizer_pid" 2>/dev/null || true
    fi
}
```

Then after the wait, remove:
```
if [[ -n "$forwarded_status" && $status -gt 128 ]]; then
    wait "$sanitizer_pid" 2>/dev/null || true
    status="$forwarded_status"
fi
```

And before the end of the file (replace `exit $status` with):

```
if [[ -n "$received_signal" ]]; then
    kill "-$received_signal" $$    # re-raise signal against ourselves.
    wait
fi

exit $status
```

File: build/sanitizer-java/bin/java (hunk 2 above)
Comment (WilliamK112) [reply build/sanitizer-java/bin/java]: Addressed in `11380031`. The wrapper now records `WORKDIR="$PWD"` once, passes an absolute `--log-file` path, and uses the same absolute directory for `find`. The fake failure-path test confirms the diagnostic header uses the absolute path and only the final 80 log lines are emitted.

File: build/sanitizer-java/bin/java (hunk 3 above)
Comment (WilliamK112) [reply build/sanitizer-java/bin/java]: Addressed in `11380031`. I replaced the hardcoded exit-status mapping with `received_signal`, forward the exact TERM/INT/HUP signal, clear the traps after `wait`, and re-raise the same signal against the wrapper before exit. Fake-wrapper tests verify TERM → 143 and HUP → 129; `bash -n` and `git diff --check` pass, and the repository ShellCheck/pre-commit jobs are running on this head.

**Comment by WilliamK112** [comment]
Thanks for the review. The two code changes are now in signed-off commit `11380031`:

- the log destination and lookup both use the absolute startup `WORKDIR`;
- TERM/INT/HUP are forwarded by name and then re-raised against the wrapper instead of being mapped to hardcoded exit codes.

Local fake-wrapper coverage verifies the success path, the failure path with only the final 80 log lines, the absolute log path, and TERM/HUP forwarding plus re-raise semantics. The repository license, signoff, ShellCheck, and Greptile checks are green on this head (pre-commit.ci is still pending at the time of this comment).

For the requested real compute-sanitizer failure output: I checked the available automation before claiming proof. The public GitHub Actions jobs do not have a GPU, and the Blossom GPU premerge path runs `ci/premerge-build.sh`, which does not enable `-DUSE_SANITIZER=ON`. My macOS host therefore cannot produce an authentic CUDA out-of-bounds report.

I can prepare a short-lived proof-only commit that enables `USE_SANITIZER` for one deliberately failing test/kernel, have a maintainer trigger the Blossom `build`, capture the resulting wrapper output for this PR, and then revert the proof-only commit before merge. Would you prefer that route, or is there a maintainer-side way to invoke the GPU job with `-DUSE_SANITIZER=ON` without changing the branch?

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
Thanks @WilliamK112 this is looking better, but I need a couple of more changes. I ran your branch locally and added a bug to cause compute-sanitizer to trigger (and it does), but the issue is that the output goes to stderr, which surefire sends to a dumpstream file and not to the terminal, so it hides away in targe/surefire-reports. 

The other issue is we are tailing the last 80 lines and really a lot of the interesting issues are at the beginning in compute-sanitizer, so I am turning that on its head, literally, and doing the first 80 lines.

This is what the current patch does, outputting to dumpstream (not ideal):

```
# Created at 2026-08-05T16:03:06.077
compute-sanitizer failed with exit code 254.

# Created at 2026-08-05T16:03:06.082
===== Last 80 lines of cudf-spark-jni-pr4752/sanitizer_for_pid_577_8679_32039_611.log =====

# Created at 2026-08-05T16:03:06.086
=========         Host Frame:  [0x7ca626d] in cudf7136924022116732976.so

# Created at 2026-08-05T16:03:06.086
=========         Host Frame: rmm::mr::cuda_memory_resource::allocate(cuda::__4::stream_ref, unsigned long, unsigned long) [0x102737b4] in cudf7136924022116732976.so

# Created at 2026-08-05T16:03:06.086
=========         Host Frame: Java_ai_rapids_cudf_Rmm_allocInternal [0x2fddc07] in cudf7136924022116732976.so
```

Here are the patches you'd need to apply to output to stdout alongside the test:

```
diff --git a/build/sanitizer-java/bin/java b/build/sanitizer-java/bin/java
index 5c30efd8..352e4f14 100755
--- a/build/sanitizer-java/bin/java
+++ b/build/sanitizer-java/bin/java
@@ -20,7 +20,7 @@
 # the tests will run with the compute-sanitizer tool.
 log_token="${BASHPID:-$$}_${RANDOM}_${RANDOM}"
 log_basename="sanitizer_for_pid_${log_token}"
-WORKDIR="$PWD"
+WORKDIR="${SANITIZER_LOG_DIR:-$PWD}"
 sanitizer_pid=""
 received_signal=""
 
@@ -48,21 +48,6 @@ wait "$sanitizer_pid"
 status=$?
 trap - TERM INT HUP
 
-if [[ $status -ne 0 ]]; then
-    echo "compute-sanitizer failed with exit code $status." >&2
-
-    found_logs=false
-    while IFS= read -r sanitizer_log; do
-        found_logs=true
-        echo "===== Last 80 lines of $sanitizer_log =====" >&2
-        tail -n 80 "$sanitizer_log" >&2
-    done < <(find "$WORKDIR" -maxdepth 1 -type f -name "${log_basename}_*.log" | sort)
-
-    if [[ $found_logs == false ]]; then
-        echo "No sanitizer_for_pid_*.log files were created for this fork." >&2
-    fi
-fi
-
 if [[ -n "$received_signal" ]]; then
     kill "-$received_signal" "$$"
     wait
```

```
diff --git a/pom.xml b/pom.xml
index 45694bf5..6b6e8c8c 100644
--- a/pom.xml
+++ b/pom.xml
@@ -256,6 +256,10 @@
                 <configuration>
                   <groups>!noSanitizer</groups>
                   <jvm>${project.basedir}/build/sanitizer-java/bin/java</jvm>
+                  <environmentVariables>
+                    <SANITIZER_LOG_DIR>${project.build.directory}/surefire-reports</SANITIZER_LOG_DIR>
+                  </environmentVariables>
+                  <testFailureIgnore>true</testFailureIgnore>
                 </configuration>
               </execution>
               <execution>
@@ -275,6 +279,49 @@
                 </goals>
                 <configuration>
                   <jvm>${project.basedir}/build/sanitizer-java/bin/java</jvm>
+                  <environmentVariables>
+                    <SANITIZER_LOG_DIR>${project.build.directory}/surefire-reports</SANITIZER_LOG_DIR>
+                  </environmentVariables>
+                  <testFailureIgnore>true</testFailureIgnore>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-antrun-plugin</artifactId>
+            <version>${antrun.version}</version>
+            <executions>
+              <execution>
+                <id>clean-sanitizer-logs</id>
+                <phase>process-test-classes</phase>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <configuration>
+                  <target>
+                    <delete>
+                      <fileset dir="${project.build.directory}/surefire-reports"
+                               includes="sanitizer_for_pid_*.log"
+                               erroronmissingdir="false"/>
+                    </delete>
+                  </target>
+                </configuration>
+              </execution>
+              <execution>
+                <id>print-sanitizer-logs</id>
+                <phase>test</phase>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <configuration>
+                  <target>
+                    <echo message="Checking for compute-sanitizer logs..."/>
+                    <exec executable="bash" failonerror="true">
+                      <arg value="-c"/>
+                      <arg value="found=0; for f in &quot;${project.build.directory}/surefire-reports&quot;/sanitizer_for_pid_*.log; do [ -f &quot;$f&quot; ] || continue; printf '===== %s =====\n' &quot;$f&quot;; head -n 80 &quot;$f&quot;; grep -q 'ERROR SUMMARY: 0 errors' &quot;$f&quot; || found=1; done; exit $found"/>
+                    </exec>
+                  </target>
                 </configuration>
               </execution>
             </executions>
```

Here is the sample output:

```
....
[ERROR] Tests run: 16, Failures: 2, Errors: 11, Skipped: 0
[INFO] 
[ERROR] There are test failures.

Please refer to cudf-spark-jni-pr4752/target/surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
Error occurred in starting fork, check output in log
org.apache.maven.surefire.booter.SurefireBooterForkException: Error occurred in starting fork, check output in log
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:624)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:954)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)

[INFO] 
[INFO] --- maven-antrun-plugin:3.0.0:run (print-sanitizer-logs) @ spark-rapids-jni ---
[INFO] Executing tasks
[WARNING]      [echo] Checking for compute-sanitizer logs...
[INFO]      [exec] ===== cudf-spark-jni-pr4752/target/surefire-reports/sanitizer_for_pid_578_29261_18771_612.log =====
[INFO]      [exec] ========= COMPUTE-SANITIZER
[INFO]      [exec] ========= Invalid __global__ write of size 4 bytes
[INFO]      [exec] =========     at void spark_rapids_jni::<unnamed>::gpu_bloom_filter_put<(int)1, (bool)0>(unsigned int *, long, cudf::column_device_view, int, int)+0x19e0
[INFO]      [exec] =========     by thread (0,0,0) in block (0,0,0)
[INFO]      [exec] =========     Access at 0x7112f288060c is out of bounds
[INFO]      [exec] =========     and is 1 bytes after the nearest allocation at 0x7112f2800600 of size 524300 bytes
[INFO]      [exec] =========     Saved host backtrace up to driver entry point at kernel launch time
[INFO]      [exec] =========         Host Frame: cuLaunchKernel_ptsz [0x33ce0a] in libcuda.so.1
[INFO]      [exec] =========         Host Frame:  [0x7c6ad95] in cudf7349655614744512972.so
[INFO]      [exec] =========         Host Frame:  [0x7cae07d] in cudf7349655614744512972.so
[INFO]      [exec] =========         Host Frame: spark_rapids_jni::bloom_filter_put(cudf::list_scalar&, cudf::column_view const&, rmm::cuda_stream_view) [0x2ad5884] in cudf7349655614744512972.so
[INFO]      [exec] =========         Host Frame: Java_com_nvidia_spark_rapids_jni_BloomFilter_put [0x2a2b20e] in cudf7349655614744512972.so
[INFO]      [exec] =========         Host Frame:  [0xffffffffe3357ab4]
[INFO]      [exec] ========= 
```

**Review by abellina — CHANGES_REQUESTED** [review]
@WilliamK112 I sent a list of patches in my previous comment that I think will let us merge this in, please take a look.

**Comment by sameerz** [comment]
@WilliamK112 in addition to @abellina's suggestions, since we are in code freeze for `release/26.08`, let's retarget this back to `main`.

**Comment by WilliamK112** [comment]
@abellina Addressed the latest maintainer requests on signed head `e05a9931` and retargeted the PR to `main`. The wrapper now writes uniquely named reports to `SANITIZER_LOG_DIR` without sending excerpts to the forked JVM stderr; Maven cleans stale reports, prints the first 80 lines of each report in the normal build output, and fails on a nonzero sanitizer summary. I also updated the Compute Sanitizer documentation. Local validation covered Bash/XML syntax, the active effective POM, fake wrapper success/failure behavior, and direct Maven `antrun:run@print-sanitizer-logs` runs for both zero-error and error reports. All available checks are green. Could you re-review when convenient?

**Comment by WilliamK112** [comment]
@abellina I found and fixed one follow-up edge case in the requested Maven flow on signed head 4d28345a: testFailureIgnore is needed so the post-test sanitizer output runs, but it could otherwise let ordinary Java test failures pass when sanitizer reports were clean. The sanitizer profile now removes stale TEST-*.xml reports before testing and, after printing sanitizer logs, also fails on failure or error elements in the current Surefire XML reports. Direct Maven fixtures verify that clean reports succeed while a JUnit failure, a JUnit error, or a nonzero sanitizer summary each fail; the cleanup execution also removes both stale report types. License, ShellCheck, signoff, and pre-commit checks are green, with Greptile still rerunning. Could you re-review this head when convenient?

**Comment by WilliamK112** [comment]
@abellina Your patches are already on this branch: sanitizer reports go to `SANITIZER_LOG_DIR` / `target/surefire-reports`, Maven prints the first 80 lines instead of dumping stderr, and the build still fails on a nonzero `ERROR SUMMARY` or Surefire `<failure>`/`<error>` XML after `testFailureIgnore`.

I merged current `main` into the branch (`f843e2b5`) so it is no longer hundreds of commits behind. Could you re-review this head when convenient?

@sameerz This remains targeted at `main`.

**Comment by nvauto** [comment]
**NOTE: release/26.10 has been created from main. Please retarget your PR to release/26.10 if it should be included in the release.**
